### PR TITLE
 android: use PermissionListener to avoid Android manual steps 

### PR DESCRIPTION
**Pull request by saghul**
See #243 and #248

### CODE REVIEW

**Comment by saghul** [comment]
Ping?

**Comment by saghul** [comment]
Gentle reminder @wmcmahan :-)

**Comment by saghul** [comment]
Ping...

**Comment by MoOx** [comment]
@saghul can you resolve conflicts please?

**Comment by saghul** [comment]
Will do!

**Comment by saghul** [comment]
@MoOx rebased, please take a look!

**Comment by saghul** [comment]
Ping again...

**Comment by saghul** [comment]
👋  About to celebrate the PRs first birthday here.

**Comment by MoOx** [comment]
I was wondering if we should handle permissions in the codebase as we have a solid react-native-permissions solution this days. Thoughts?

**Comment by saghul** [comment]
@MoOx I don't know what you mean. This change uses that exact permissions solution, the builtin one. No glue code is necessary to make it work, it piggybacks on the same mechanism used by the PermissionsAndroid module, for instance. What do you suggest?

**Comment by MoOx** [comment]
@saghul I mean we could remove all the "permission" check/request in the codebase of this project & rely on more solid ones like react-native-permissions that can now ask only for a single permissions (without bloating your Info.plist) for iOS.

**Comment by saghul** [comment]
Why spend the time doing that when this plugin works just fine as is? This PR is a net improvement for Android, because it requires users to do one less step.

I've been using our fork for quite a while now on an app installed by millions of users with no issues.

I don't want to sound rude, but it's really discouraging to have this un-merged for so long.

**Comment by MoOx** [comment]
Hi @saghul. I feel your frustration. Sorry but I am not the original author of this lib & just tried to help a little. I asked for write access so I can merge PR & triage issues but I don't feel confortable enough with the existing code to feel confident when merging stuff :)
You seems to be a good open source contributor, would you be interested to help me maintaining this lib? I can ask original author to add you as a maintainer so we can move forward with it (faster than I do know lol)

**Comment by saghul** [comment]
@MoOx Sounds good to me! Thanks for understanding!

**Comment by MoOx** [comment]
@saghul you should have received an invite!

**Comment by MoOx** [comment]
I will let you merge your own PR :) (please use Squash & Merge to keep the git history concise, which helps for changelog).
I am having a hard time with #296 & would like to cut a release after that & this PR (& maybe other if you have time to review some)

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-native-calendar-events",
-  "version": "1.7.3",
+  "version": "2.0.0",
```
Comment (MoOx) [inline package.json:3]: Maybe revert this so we can bump the version only when we cut the release?

**Comment by MoOx** [comment]
Cleaning up the repo a little bit, and being more familiar with it, this PR makes TOTAL SENSE. Thanks @saghul !
Do not hesitate to help more :p

**Pull request merged**